optimum/onnxruntime/trainer.py

```diff
@@ -970,10 +979,16 @@ def evaluation_loop_ort(
         else:
             ort_model_cls = ORTModelForCustomTasks
 
-        model_id, file_name = os.path.split(self.onnx_model_path)
-        ort_model = ort_model_cls.from_pretrained(model_id=model_id, file_name=file_name)
-
+        model_id = self.onnx_model_path
         args = self.args
+        # Temporary fix for decoder, now `use_cache` set to False which
+        # TODO: Use cache once `ORTModelForCausalLM` supports `loss` as output
+        if ort_model_cls == ORTModelForCausalLM:
+            ort_model = ort_model_cls.from_pretrained(
+                model_id=model_id, use_cache=False, provider="CUDAExecutionProvider"
+            )
+        else:
+            ort_model = ort_model_cls.from_pretrained(model_id=model_id, provider="CUDAExecutionProvider")
```

Comment:
```suggestion
       ort_model = ort_model_cls.from_pretrained(
                model_id=model_id, use_cache=ort_model_cls is not ORTModelForCausalLM, provider="CUDAExecutionProvider"
            )
```

Reply:
Other subclasses of `ORTModel` don't have `use_cache` argument will fail.